flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpanningWrapper.java

```diff
@@ -286,11 +292,21 @@ private FileChannel createSpillingChannel() throws IOException {
 		// try to find a unique file name for the spilling channel
 		int maxAttempts = 10;
 		for (int attempt = 0; attempt < maxAttempts; attempt++) {
-			String directory = tempDirs[rnd.nextInt(tempDirs.length)];
+			int dirIndex = rnd.nextInt(tempDirs.length);
+			String directory = tempDirs[dirIndex];
 			File file = new File(directory, randomString(rnd) + ".inputchannel");
-			if (file.createNewFile()) {
-				spillFile = new RefCountedFile(file);
-				return new RandomAccessFile(file, "rw").getChannel();
+			try {
+				if (file.createNewFile()) {
+					spillFile = new RefCountedFile(file);
+					return new RandomAccessFile(file, "rw").getChannel();
+				}
+			} catch (IOException e) {
+				// if there is no tempDir left to try
+				if (tempDirs.length <= 1) {
+					throw e;
+				}
+				LOG.warn("Caught an IOException when creating spill file: " + directory + ". Attempt " + attempt, e);
+				tempDirs = (String[]) ArrayUtils.remove(tempDirs, dirIndex);
```

Comment:
If we settle on such trivial approach to the problem (without temporary blacklisting), I wouldn't remove the failed dir from the `tempDir`, but just keep it and re-try next time?

Reply:
I did the same as you suggest, and it probably leads to too many "WARNING" logs.

Reply:
Or we just remove the ”LOG.warn“ here?

Reply:
Ok, good point. Let's keep it as it is. At worst if there are some intermittent errors and we deplete the available temp dirs, job will restart and re attempt to write to the same directories (and the intermittent error might be gone by then).

Reply:
Agree with that.